### DISCUSSION TITLE
More handlebars helpers? Split in particular.  Can we register new helpers on the fly?

### DISCUSSION THREAD

**Discussion by proCrichard** [renovatebot/renovate#21787] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovatebot/github-action@v36.0.2

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Using Renovate on a gitops repo to keep both container images and helm charts up to date.

Many of our helm releases, managed by kustomize, are deployed using the same helm chart, and so renovate tends to group them on the dependency dashboard/PRs, which we want to avoid.  Luckily I can split these packages up using a provided template field in a packageRule:

```
"commitMessagePrefix": "{{packageFileDir}}",
"additionalBranchPrefix": "{{packageFileDir}}-",
```

Great!  Problem is now the result on the dependency dashboard is really annoyingly long and hard to parse.  Kind of like:

`apps/tenant1/my-app/production/cluster-a - Update Helm release my-chart to v0.3.22`

It would be nice to split off just the `my-app` segment of the path, or any other segment I want, for the commit message.  The solution seems simple to me, use a [handlebars helper](https://github.com/helpers/handlebars-helpers#split)!

`"commitMessagePrefix": "{{#with (split packageFileDir \"/\")}}{{lookup this 2}}{{/with}}"`

Unfortunately this triggers the following error:

```
"err": {
    "message": "Missing helper: \"split\"",
    ...
```

### What you tried so far.

It seems there is no way to register custom handlebars helpers as part of configuration when running renovate.  It would be very helpful to be able to pass in a `helpers.js` file to renovate when starting up the bot to register custom helpers.

Alternatively if some of the most [useful helpers](https://github.com/helpers/handlebars-helpers) (split in particular) could be registered [here](https://github.com/renovatebot/renovate/blob/4dcf521a9a18759766e56decdf6760758bed7e50/lib/util/template/index.ts) it would solve my problem.

**Comment by viceice** [comment]
use the replace helper, it supports regEx

**Reply by proCrichard** [reply]
This seems like not what the replace helper was intended for.
I don't want to replace any of the portions of the path in `packageFileDir`, just pull strings from it.

The regEx I would need to provide to the replace helper to get that to work seems way too complex ~(not possible?)~ when adding the split helper would be a vastly more simple.  ~Not only that but I'd need to write a unique regex for each application in our repo, when it could be done dynamically with the split helper.~ not true. replace does work in my context. see below.

**Reply by proCrichard** [reply]
I did get replace to work, but I think this looks unreadable:

```
{{{replace '^.*?\/.*?\/([A-Za-z0-9&-]+)\/.*$' '$1' packageFileDir}}}
```
I'm going to leave this discussion open as I think the option to register custom helpers for handlebars somehow would be nicer.
